feat: Return resource id on successful creation from `Location` header

Star3Lord opened the pull request.

kilork commented:
Hi, thanks for this idea.

Some notes from my side, as currently this PR could not be merged:

- The code is generated in `openapi.rs`, we do not write it manually;
- Could we use `TypeString` directly without `into`?
- Why `unwrap` instead of `?`?
- Why not use header constants, like `reqwest::header::CONTENT_LENGTH`?

Star3Lord commented:
> Hi, thanks for this idea.
> 
> Some notes from my side, as currently this PR could not be merged:
> 
> * The code is generated in `openapi.rs`, we do not write it manually;
> * Could we use `TypeString` directly without `into`?
> * Why `unwrap` instead of `?`?
> * Why not use header constants, like `reqwest::header::CONTENT_LENGTH`?

- Made updates to `openapi.rs` to generate the changes.
- I updated the `get_id_from_location_header` function in `src/rest/mod.rs` but return `Options<TypeString>` now.
- Removed `unwrap()`. I am not sure if it should error or not, if url to newly created resource is present, it should return that, otherwise nothing(as was the case before this pr). So, `Option` makes sense in my opinion.
- Done, now using `reqwest::header::LOCATION`

kilork commented:
Okay, after small corrections and *actually* running `openapi.rs` generator - going to merge it. Thanks for this contribution!
